core/src/main/java/org/geysermc/geyser/session/GeyserSession.java

```diff
@@ -687,63 +685,42 @@ public void authenticateWithMicrosoftCode() {
         packet.setTime(16000);
         sendUpstreamPacket(packet);
 
-        // new thread so clients don't timeout
-        MsaAuthenticationService msaAuthenticationService = new MsaAuthenticationService(GeyserImpl.OAUTH_CLIENT_ID);
+        final PendingMicrosoftAuthentication.AuthenticationTask task = geyser.getPendingMicrosoftAuthentication().getOrCreateTask(
+                Objects.requireNonNull(getAuthData(), "authData").xuid()
```

Comment:
I think the requireNonNull here is unnecessary - if authData is null, then when we load the player's locale we're going to get a NullPointerException.